tests/posix/common/src/pthread.c

```diff
@@ -539,6 +539,35 @@ ZTEST(pthread, test_pthread_testcancel)
 	zassert_false(testcancel_failed);
 }
 
+static void *test_pthread_setschedprio_fn(void *arg)
+{
+	pthread_t self;
+	int prio = K_LOWEST_APPLICATION_THREAD_PRIO;
+	int policy;
+	struct sched_param getschedparam;
+
+	getschedparam.sched_priority = K_HIGHEST_APPLICATION_THREAD_PRIO;
+
+	self = pthread_self();
+
+	zassert_equal(pthread_setschedprio(self, PRIO_INVALID), EINVAL, "EINVAL was expected");
+	zassert_equal(pthread_setschedprio(PTHREAD_INVALID, prio), ESRCH, "ESRCH was expected");
+
+	zassert_ok(pthread_setschedprio(self, prio));
+	zassert_ok(pthread_getschedparam(self, &policy, &getschedparam));
```

Comment:
It would be good to change `getschedparam.sched_priority` (maybe bitwise inversion `~`) prior to this call.

Reply:
I actually do that with :
`getschedparam.sched_priority = K_HIGHEST_APPLICATION_THREAD_PRIO;`
(but no bitwise inversion ^^)